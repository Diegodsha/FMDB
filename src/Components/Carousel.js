/* eslint-disable no-unused-vars */
import { useEffect, useState } from 'react';
import axios from 'axios';
import { BASE_URL } from '../API/api';
import { moviesEndPoint } from '../API/Endpoints';
import StyledCarousel from '../Styles/Carrousel.module.css';

const Carousel = () => {
  const [FivePopularMovies, setFivePopularMovies] = useState([]);
  useEffect(() => {
    axios.get(`${BASE_URL}${moviesEndPoint.mostPopular}`).then((res) => {
      setFivePopularMovies(res.data.results.slice(0, 5));
    });
  }, []);

  return (
    <div
      id="carouselExampleCaptions"
      className={`carousel slide ${StyledCarousel.margin}`}
      data-bs-ride="carousel"
    >
      <div className="carousel-indicators">
        <button
          type="button"
          data-bs-target="#carouselExampleCaptions"
          data-bs-slide-to="0"
          className="active"
          aria-current="true"
          aria-label="Slide 1"
        />
        <button
          type="button"
          data-bs-target="#carouselExampleCaptions"
          data-bs-slide-to="1"
          aria-label="Slide 2"
        />
        <button
          type="button"
          data-bs-target="#carouselExampleCaptions"
          data-bs-slide-to="2"
          aria-label="Slide 3"
        />
        <button
          type="button"
          data-bs-target="#carouselExampleCaptions"
          data-bs-slide-to="3"
          aria-label="Slide 4"
        />
        <button
          type="button"
          data-bs-target="#carouselExampleCaptions"
          data-bs-slide-to="4"
          aria-label="Slide 5"
        />
      </div>
      <div className={`carousel-inner ${StyledCarousel.rounded}`}>
        <div className="carousel-item active">
          <img
            src={`https://image.tmdb.org/t/p/original${FivePopularMovies[0]?.poster_path}`}
            className={`d-block w-100 ${StyledCarousel.fit}`}
            alt="..."
          />
        </div>
        {FivePopularMovies.slice(1)?.map((item) => (
          <div key={item.id} className="carousel-item">
            <img
              src={`https://image.tmdb.org/t/p/original${item?.poster_path}`}
              className={`d-block w-100 ${StyledCarousel.fit}`}
              alt={item?.title}
            />
          </div>
        ))}
      </div>
      <button
        className="carousel-control-prev"
        type="button"
        data-bs-target="#carouselExampleCaptions"
        data-bs-slide="prev"
      >
        <span className="carousel-control-prev-icon" aria-hidden="true" />
        <span className="visually-hidden">Previous</span>
      </button>
      <button
        className="carousel-control-next"
        type="button"
        data-bs-target="#carouselExampleCaptions"
        data-bs-slide="next"
      >
        <span className="carousel-control-next-icon" aria-hidden="true" />
        <span className="visually-hidden">Next</span>
      </button>
    </div>
  );
};

export default Carousel;
