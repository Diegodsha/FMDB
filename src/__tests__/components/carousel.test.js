import renderer from 'react-test-renderer';
import Carousel from '../../Components/Carousel';

describe('Carousel Component', () => {
  test('Renders correctly component', () => {
    const tree = renderer.create(<Carousel />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
