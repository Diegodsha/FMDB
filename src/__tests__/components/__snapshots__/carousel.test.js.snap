// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Component Renders correctly component 1`] = `
<div
  className="carousel slide margin"
  data-bs-ride="carousel"
  id="carouselExampleCaptions"
>
  <div
    className="carousel-indicators"
  >
    <button
      aria-current="true"
      aria-label="Slide 1"
      className="active"
      data-bs-slide-to="0"
      data-bs-target="#carouselExampleCaptions"
      type="button"
    />
    <button
      aria-label="Slide 2"
      data-bs-slide-to="1"
      data-bs-target="#carouselExampleCaptions"
      type="button"
    />
    <button
      aria-label="Slide 3"
      data-bs-slide-to="2"
      data-bs-target="#carouselExampleCaptions"
      type="button"
    />
    <button
      aria-label="Slide 4"
      data-bs-slide-to="3"
      data-bs-target="#carouselExampleCaptions"
      type="button"
    />
    <button
      aria-label="Slide 5"
      data-bs-slide-to="4"
      data-bs-target="#carouselExampleCaptions"
      type="button"
    />
  </div>
  <div
    className="carousel-inner rounded"
  >
    <div
      className="carousel-item active"
    >
      <img
        alt="..."
        className="d-block w-100 fit"
        src="https://image.tmdb.org/t/p/originalundefined"
      />
    </div>
  </div>
  <button
    className="carousel-control-prev"
    data-bs-slide="prev"
    data-bs-target="#carouselExampleCaptions"
    type="button"
  >
    <span
      aria-hidden="true"
      className="carousel-control-prev-icon"
    />
    <span
      className="visually-hidden"
    >
      Previous
    </span>
  </button>
  <button
    className="carousel-control-next"
    data-bs-slide="next"
    data-bs-target="#carouselExampleCaptions"
    type="button"
  >
    <span
      aria-hidden="true"
      className="carousel-control-next-icon"
    />
    <span
      className="visually-hidden"
    >
      Next
    </span>
  </button>
</div>
`;
